quilt_lite/quilt_lite/s3proxy.py

```diff
@@ -0,0 +1,74 @@
+"""
+Python implementation of the s3 proxy from quilt/s3-proxy/nginx.conf
+"""
+import urllib.parse
+
+import aiohttp
+from fastapi import FastAPI, Request, Response
+
+
+s3proxy = FastAPI()
+
+session = aiohttp.ClientSession()
+
+
+@s3proxy.api_route("/{s3_region}/{s3_bucket}", methods=['GET', 'HEAD', 'POST', 'PUT', 'OPTIONS'])
+@s3proxy.api_route("/{s3_region}/{s3_bucket}/{s3_path:path}", methods=['GET', 'HEAD', 'POST', 'PUT', 'OPTIONS'])
+async def s3proxy_request(request: Request, s3_region: str, s3_bucket:str, s3_path: str = ''):
+    """
+    Forward the request to S3.
+    """
+    cors_headers = {
+        'access-control-allow-headers': request.headers.get('access-control-request-headers', ''),
+        'access-control-allow-methods': request.headers.get('access-control-request-method', ''),
+        'access-control-allow-origin': '*',
+        'access-control-max-age': '3000',
+        'access-control-expose-headers': ', '.join([
+            'Content-Length',
+            'Content-Range',
+            'ETag',
+            'x-amz-bucket-region',
+            'x-amz-delete-marker',
+            'x-amz-request-id',
+            'x-amz-version-id',
+            'x-amz-storage-class',
+        ])
+    }
+
+    req_body = await request.body()
+
+    if request.method == 'OPTIONS':
+        return Response(content='', status_code=200, headers=cors_headers)
+
+    if s3_region == '-':
```

Comment:
Is `-` an AWS thing or just a convention we use?

Reply:
Our convention. It was added in #1888; I forgot exactly why - it might have been only for Zip downloads - unless the catalog uses it too somewhere (@nl0 ?)

But, regardless, S3 has two URL formats, with and without a region, so it'll come up sooner or later.

Reply:
i don't remember exactly, but i think we use regionless requests a lot